addons/google_account/controllers/main.py

```diff
@@ -13,11 +14,13 @@ class GoogleAuth(http.Controller):
     @http.route('/google_account/authentication', type='http', auth="public")
     def oauth2callback(self, **kw):
         """ This route/function is called by Google when user Accept/Refuse the consent of Google """
-        state = json.loads(kw['state'])
+        state = json.loads(kw.get('state', '{}'))
         dbname = state.get('d')
         service = state.get('s')
         url_return = state.get('f')
         base_url = request.httprequest.url_root.strip('/')
+        if (not dbname or not service or (kw.get('code') and not url_return)):
```

Comment:
```suggestion
if not dbname or not service or not url_return:
```
this would seem enough to me, since the 3 code branches are using url_return